draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1066,182 @@ The following describes each member of the `concise-bom-tag` map.
   The `$$concise-bom-tag-extension` extension socket is empty in this
   specification.
 
+# CoRIM-based Evidence Verification
+
+The verification procedure is divided into three separate phases:
+
+* Appraisal Context initialisation
+* Evidence collection
+* Evidence appraisal
+
+At a few well-defined points in the procedure, the Verifier behaviour will depend on the specific CoRIM profile.
+Each CoRIM profile SHALL fully describe the expected Verifier behaviour for each of those well-defined points.
+
+Note that what follows is not meant to describe a real implementation.
+In particular, it is expected that the cost associated with the initialisation phase can be amortised across multiple appraisals.
+Implementers are free to do as they please, as long as the stated invariants are not broken.
+
+In the following, if a MUST-level requirement is violated, the entire procedure is aborted.
+
+## Appraisal Context initialisation
+
+The goal of the initialisation phase is to load the CoRIM Appraisal Context with objects such as CoRIM tags, cryptographic validation key material (e.g., raw public keys, root certificates, intermediate CA certificate chains, etc.) etc. that will be used in the subsequent Evidence Appraisal phase.
+
+### CoRIM Selection
+
+All available Concise Reference Integrity Manifests (CoRIMs) are collected. A verifier may be pre-configured with a large number of tags describing many types of device. All CoRIMs are loaded at this stage, later stages will select the CoRIMs appropriate to the evidence appraisal step.
+
+CoRIMs that are not within their validity period, or that cannot be associated with an authenticated and authorised source MUST be discarded.
+
+CoRIM that are secured by a cryptographic mechanism such as a signature MUST be discarded if the signature chain does not validate successfully.
+
+Other selection criteria MAY be applied.
+For example, if the Evidence format is known in advance, CoRIMs that do not match the expected profile can be readily discarded.
+
+The selection process MUST yield at least one usable CoRIM.
+
+### CoBOMs Extraction
+
+All the available Concise Bill Of Material tags (CoBOMs) are collected from the selected CoRIMs.
+
+If any CoBOM is present then other tags in the CoRIM which are not activated by the CoBOM MUST be ignored.
```

Comment:
CDDL for `corim-map` doesn't make use of CoBOM tag mandatory. But the prose makes it mandatory - see [1]. Given [1], the language in 1107 can describe 'unactivated tags' as ignored. Not sure it needs to be normative MUST since [1] already contains normative. 

[1]
A Concise Bill of Material (CoBOM) object represents the signal for the verifier to activate the listed tags. Data contained in a tag MUST NOT be used for appraisal until a CoBOM which activates that tag has been received and successfully processed. All the tags listed in the CoBOM must be activated in the same transaction, i.e., either all or none.